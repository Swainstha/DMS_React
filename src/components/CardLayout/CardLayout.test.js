import React from 'react';
import CardLayout from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const setup = (props = {}) => {
    return shallow( < CardLayout {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


it("CardLayout",()=>{
    const wrapper= setup({children:(<div></div>)});
    expect(wrapper).toMatchSnapshot();
})

it("CardLayout props",()=>{
    const wrapper= setup({children:[<div data-test="child0" key ="child0"></div>,<div key="child1" data-test="child1"></div>]});

    let child0= findByTestAttr(wrapper, 'child0')
    let child1= findByTestAttr(wrapper, 'child1')
    let div_children = wrapper.find('div')
    expect(div_children).toHaveLength(3)
    expect(child0).toHaveLength(1)
    expect(child1).toHaveLength(1)

})