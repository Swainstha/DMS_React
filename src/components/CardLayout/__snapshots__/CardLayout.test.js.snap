// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLayout 1`] = `ShallowWrapper {}`;
